connectors/src/connectors/notion/temporal/workflows/deletion_crawl.ts

```diff
@@ -62,20 +62,19 @@ export async function notionDeletionCrawlWorkflow({
     }
 
     while (resourceQueue.length > 0) {
-      const resource = resourceQueue.shift();
-      if (!resource) {
+      const resources = resourceQueue.splice(0, DELETE_BATCH_SIZE);
+      if (resources.length === 0) {
```

Comment:
So this should never happen, right? We got into the while because it was non-empty.